packages/flutter_tools/lib/src/commands/packages.dart

```diff
@@ -417,4 +417,35 @@ class PackagesGetCommand extends FlutterCommand {
       commandPackagesAndroidEmbeddingVersion: rootProject.android.getEmbeddingVersion().toString().split('.').last,
     );
   }
+
+  /// The pub packages usage values are incorrect since these are calculated/sent
+  /// before pub get completes. This needs to be performed after dependency resolution.
+  @override
+  Future<Event> unifiedAnalyticsUsageValues(String commandPath) async {
+    final FlutterProject? rootProject = _rootProject;
+    if (rootProject == null) {
+      return Event.commandUsageValues(workflow: commandPath, commandHasTerminal: hasTerminal);
+    }
+
+    int numberPlugins;
+    // Do not send plugin analytics if pub has not run before.
+    final bool hasPlugins = rootProject.flutterPluginsDependenciesFile.existsSync()
+      && rootProject.packageConfigFile.existsSync();
+    if (hasPlugins) {
+      // Do not fail pub get if package config files are invalid before pub has
+      // had a chance to run.
+      final List<Plugin> plugins = await findPlugins(rootProject, throwOnError: false);
```

Comment:
i bet this could be slow, can we cache this on `class PackagesGetCommand` so we don't have to repeat the work?